src/util/fileinfo.h

```diff
@@ -0,0 +1,304 @@
+#pragma once
+
+#include <gtest/gtest_prod.h>
+
+#include <QDateTime>
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QUrl>
+#include <QtDebug>
+
+#include "util/assert.h"
+
+namespace mixxx {
+
+/// Helper functions and extensions for QFileInfo using Mixxx terminology.
+namespace fileinfo {
```

Comment:
How would you then name those static functions to resolve the naming conflicts?

Reply:
We can have a static and a non static function with the same name, because the parameters are different.

Reply:
```
QSting location = AbsoluteFileInfo::location(file);
``` 

Looks Ok to me.

The purpose of the assert inside would be obvious. No need to a extra comment.

Reply:
The property of being *not relative* is not a class invariant. Instead this must be determined at runtime whenver you try to obtain a *location* string.

Reply:
How could it happen that it changes after the QFileObject is constructed?
I am unsure how this relates to my interface proposal, because it just showsa rename and a move into the class namespace.

Reply:
Please note that only some `mixxx::FileInfo` instances will be converted into location strings. This must only be checked when you try to obtain a location, not in the constructor.

Reply:
> I have left some comments.
> I have not yet get the point where it helps in:
> #2656
> Can you print me to a place?

The subsequent commits in #2656 will make use of this class.